cmds/modules/noded/main.go

```diff
@@ -193,6 +195,30 @@ func action(cli *cli.Context) error {
 		return hypervisor, nil
 	})
 
+	log.Info().Msg("start perf scheduler")
+
+	perfMon, err := perf.NewPerformanceMonitor(msgBrokerCon)
+	if err != nil {
+		return errors.Wrap(err, "failed to create a new perfMon")
+	}
+
+	if err = perfMon.Run(ctx); err != nil {
+		return errors.Wrap(err, "failed to run the scheduler")
+	}
+	bus.WithHandler("zos.perf.get", func(ctx context.Context, payload []byte) (interface{}, error) {
+		var taskName string
+		err := json.Unmarshal(payload, &taskName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to unmarshal payload: %v", payload)
+		}
+
+		if taskName == "" {
+			return perfMon.GetAll()
+		}
+
+		return perfMon.Get(taskName)
```

Comment:
While this will work perfectly will in Go . I don't like the fact that the output type changes based on the input (one returns a single object, the other returns a list) which makes it annoying to call from languages that supports static type.

This can be solved in 2 possible ways
- have 2 functions `zos.perf.get` and `zos.perf.get_all` 
- Return a unified type, say list of tests which can be a list of a single test if specific name is given, or all.

A 3rd solution would be always return ALL? and then client can filter the test on client side is also a possible option